Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,341 @@
+        for (auto collider : link->collision_array)
+        { // one or more colliders - the array is used
+            AddCollider(collider, entityId, PrefabMakerUtils::MakeEntityName(link->name.c_str(), typeString, nameSuffixIndex));
+            nameSuffixIndex++;
+        }
+
+        if (link->collision_array.empty())
+        { // no colliders in the array - zero or one in total, the element member is used instead
+            AddCollider(link->collision, entityId, PrefabMakerUtils::MakeEntityName(link->name.c_str(), typeString));
+        }
+    }
+
+    void CollidersMaker::AddCollider(urdf::CollisionSharedPtr collision, AZ::EntityId entityId, const AZStd::string& generatedName)
+    {
+        if (!collision)
+        { // it is ok not to have collision in a link
+            return;
+        }
+        AZ_TracePrintf("AddCollider", "Processing collisions for entity id:%s", entityId.ToString().c_str());
+
+        auto geometry = collision->geometry;
+        if (!geometry)
+        { // non-empty visual should have a geometry
+            AZ_Warning("AddCollider", false, "No Geometry for a collider");
+            return;
+        }
+
+        AddColliderToEntity(collision, entityId);
+    }
+
+    void CollidersMaker::AddColliderToEntity(urdf::CollisionSharedPtr collision, AZ::EntityId entityId)
+    {
+        // TODO - we are unable to set collider origin. Sub-entities don't work since they would need to parent visuals etc.
+        // TODO - solution: once Collider Component supports Cylinder Shape, switch to it from Shape Collider Component.
+
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        auto geometry = collision->geometry;
+        bool isPrimitiveShape = geometry->type != urdf::Geometry::MESH;
+        if (!isPrimitiveShape)
+        {
+            entity->CreateComponent<PhysX::EditorColliderComponent>();
+            entity->Activate();
+
+            AZ_Warning("CollisionMaker", false, "Adding mesh collider");
+            auto meshGeometry = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+            auto azMeshPath = GetFullURDFMeshPath(AZ::IO::Path(m_modelPath), AZ::IO::Path(meshGeometry->filename.c_str()));
+
+            // Get asset path relative to watch folder
+            bool assetFound = false;
+            AZ::Data::AssetInfo assetInfo;
+            AZStd::string watchDir;
+            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+                assetFound,
+                &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourcePath,
+                azMeshPath.c_str(),
+                assetInfo,
+                watchDir);
+
+            // We are expecting .pxmesh
+            auto pxmodelPath = AZStd::string(assetInfo.m_relativePath.c_str());
+            AzFramework::StringFunc::Path::ReplaceExtension(pxmodelPath, ".pxmesh");
```

Comment:
To add some more context: We've been talking a lot about asset references recently on the asset pipeline team. There are several problems in this area, but the core of it is path references to assets can be very unstable.

I've been working on updated documentation to try to make it easier to understand how to manage assets referencing other assets, you can see the pull request here that I'm hoping to get in soon https://github.com/o3de/o3de.org/pull/1882

Reply:
Could you, at this moment, suggest a better approach?

Reply:
Is there where you're retrieving the FBX filename to try to extract for the physics mesh?
`auto azMeshPath = GetFullURDFMeshPath(AZ::IO::Path(m_modelPath), AZ::IO::Path(meshGeometry->filename.c_str()));`

If so, I would say either update this to point at the product asset of the physx mesh instead of the source asset of the FBX file, or if you need it for a few reasons, then add additional user data of the physx mesh to use. The URDF author is going to need to pick the correct mesh to set in the URDF file in case the FBX file has multiple physx mesh product assets, and/or the physx mesh product asset doesn't match the filename of the FBX.

Reply:
I believe that for URDF `<collider>` and `<visual>` tag only a file with a single mesh is allowed. Each link can have multiple  of these tags and this is how multiple colliders/visuals per part are handled.
Let me double check that.
Note that we will be typically working with STL or DAE files, not FBX.

Reply:
I will resolve this conversation, please continue under #156